sunpy/coordinates/transformations.py

```diff
@@ -526,7 +526,9 @@ def hgs_to_hgs(from_coo, to_frame):
     """
     Convert between two Heliographic Stonyhurst frames.
     """
-    if np.all(from_coo.obstime == to_frame.obstime):
+    if to_frame.obstime is None:
+        return from_coo.replicate()
```

Comment:
what's the difference between `.replicate()` here and `_replicate()` used in a bunch of other places?

Reply:
`._replicate()` allows one to replace the data and attributes at the same time.  `.replicate()` replaces attributes but keeps the data, while `.realize_frame()` replaces data but keeps the attributes.  Both `.replicate()` and `.realize_frame()` call `._replicate()`.

Reply:
nice and simple then :rofl: